libdevcore/CommonData.cpp

```diff
@@ -29,26 +29,52 @@
 using namespace std;
 using namespace dev;
 
+namespace
+{
+
+static char const* upperHexChars = "0123456789ABCDEF";
+static char const* lowerHexChars = "0123456789abcdef";
+
+}
+
+string dev::toHex(uint8_t _data, HexCase _case)
+{
+	assertThrow(_case != HexCase::Mixed, BadHexCase, "Mixed case can only be used for byte arrays.");
+
+	char const* chars = _case == HexCase::Upper ? upperHexChars : lowerHexChars;
+
+	return std::string{
+		chars[(unsigned(_data) / 16) & 0xf],
+		chars[unsigned(_data) & 0xf]
```

Comment:
Why cast to unsigned when the input is `uint8_t`? Also why not shift by 4 instead of divide by 16?

Reply:
I'd expect any of that to result in the same code after compiling anyways...

Reply:
Yes, but readability is different :wink

Reply:
I somehow remembered that `uint8_t`, aka `char` might be weird towards some operators on some platforms. But I could also be wrong about that...